src/app/core/services/filter/filter.service.spec.ts

```diff
@@ -22,59 +27,71 @@ describe('Filter Service', () => {
         id: 'test',
         displayType: 'text_clear',
         selectionType: 'single',
-        facets: [
-          { name: 'a', link: { uri: 'domain/filters/dahinter;SearchParameter=' } },
-          { name: 'b', link: { uri: 'domain/filters/dahinter;SearchParameter=' } },
+        filterEntries: [
+          { name: 'a', link: { uri: 'domain/productfilters/dahinter?SearchParameter=' } },
+          { name: 'b', link: { uri: 'domain/productfilters/dahinter?SearchParameter=' } },
         ],
       },
     ],
   } as FilterNavigationData;
+  let store$: StoreWithSnapshots;
 
   beforeEach(() => {
     apiService = mock(ApiService);
 
     TestBed.configureTestingModule({
-      providers: [FilterService, { provide: ApiService, useFactory: () => instance(apiService) }, provideMockStore()],
+      imports: [CoreStoreModule.forTesting(), RouterTestingModule, ShoppingStoreModule.forTesting('productListing')],
+      providers: [
+        FilterService,
+        { provide: ApiService, useFactory: () => instance(apiService) },
+        provideStoreSnapshots(),
```

Comment:
I don't see you using the store snapshots here. I think you can just use the mock store with a selector override for setProductListingPageSize.